python/ml4ir/base/pipeline.py

```diff
@@ -265,9 +318,128 @@ def get_relevance_model(self, feature_layer_keys_to_fns={}) -> RelevanceModel:
         """
         raise NotImplementedError
 
+    def kfold_analysis(self, base_logs_dir, run_id, num_folds):
+        """
+        Aggregate results of the k-fold runs and perform t-test on the results.
+        """
+        raise NotImplementedError
+
+    def create_pipeline_for_kfold(self, args):
+        raise NotImplementedError
+
+    def kfold_analysis(self, base_logs_dir, run_id, num_folds, metrics=None):
+        """
+        Aggregate results of the k-fold runs and perform t-test on the results between old(prod model) and
+        new model's w.r.t the specified metrics.
+
+        Parameters
+        ----------
+        base_logs_dir : int
+            Total number of folds
+        run_id : int
+            current fold number
+        num_folds: int
+            Total number of folds
+        metrics: set()
+            Set of metrics to include in the kfold analysis
+        """
+        if not metrics:
+            return
```

Comment:
Shouldn't we throw an error or log a warning here?

Reply:
I think we can throw a warning here